ext/windows/service/daemon.rb

```diff
@@ -54,25 +54,33 @@ def service_main(*argsv)
     #   )
     # end
 
-    puppet = File.join(basedir, 'bin', 'puppet.bat')
+    puppet = File.join(basedir, 'puppet', 'bin', 'puppet')
+    ruby = File.join(basedir, 'puppet', 'bin', 'ruby.exe')
+    ruby_puppet_cmd = "\"#{ruby}\" \"#{puppet}\""
+
     unless File.exist?(puppet)
       log_err("File not found: '#{puppet}'")
       return
     end
     log_debug("Using '#{puppet}'")
 
     cmdline_debug = argsv.index('--debug') ? :debug : nil
-    @loglevel = parse_log_level(puppet, cmdline_debug)
+    @loglevel = parse_log_level(ruby_puppet_cmd, cmdline_debug)
     log_notice('Service started')
 
     service = self
     @run_thread = Thread.new do
       begin
         while service.running? do
-          runinterval = service.parse_runinterval(puppet)
+          runinterval = service.parse_runinterval(ruby_puppet_cmd)
           if service.state == RUNNING or service.state == IDLE
+            ENV['SSL_CERT_FILE'] = File.join(basedir, 'puppet', 'ssl', 'cert.pem')
+            ENV['SSL_CERT_DIR'] = File.join(basedir, 'puppet', 'ssl', 'certs')
+            ENV['OPENSSL_CONF'] = File.join(basedir, 'puppet' ,'ssl', 'openssl.cnf')
+            ENV['FACTER_env_windows_installdir'] = basedir
```

Comment:
Do we end up with the same style of slashes here vs. the old variant?  You can use a `notify` in a manifest to emit the Facter value.

Reply:
Based on logs from CI runs, the original code resolves the value like this:

```
Debug: Facter: setting fact "env_windows_installdir" based on the value of environment variable "FACTER_env_windows_installdir".
Debug: Facter: fact "env_windows_installdir" has resolved to "C:\Program Files\Puppet Labs\Puppet".
```

I believe we'll have to vet this change in the service manually, because all of the CI runs call `puppet agent` rather than relying on the service.

Reply:
It may make sense to set this value inside of the Puppet bootstrapping itself for consistency (and so that we see the actual value in CI logs).

I know the value of the fact is consumed in `defaults.rb` like  `Facter.value(:env_windows_installdir)` at https://github.com/puppetlabs/puppet/blob/ca796215b95825f22b0ea30f56a27af181eda53a/lib/puppet/defaults.rb#L50

Thoughts @joshcooper?

Reply:
I'm not sure if a custom fact would always be resolved when running standalone `facter`? I think ideally it would just be a core fact on windows (along with other useful windows directories).